contracts/sol6/Dao/EmergencyFeeHandler.sol

```diff
@@ -0,0 +1,207 @@
+        uint256 _rebateBps,
+        uint256 _burnBps
+    ) public PermissionGroupsNoModifiers(admin) {
+        require(_burnBps.add(_rewardBps).add(_rebateBps) == BPS, "Bad BRR values");
+        rewardBps = uint16(_rewardBps);
+        rebateBps = uint16(_rebateBps);
+        kyberNetwork = _kyberNetwork;
+    }
+
+    modifier onlyKyberNetwork {
+        require(msg.sender == address(kyberNetwork), "only kyberNetwork");
+        _;
+    }
+
+    /// @dev handleFees function is called per trade on KyberNetwork. unless the trade is not involving any fees.
+    /// @param token Token currency of fees
+    /// @param rebateWallets a list of rebate wallets that will get rebate for this trade.
+    /// @param rebateBpsPerWallet percentage of rebate for each wallet, out of total rebate.
+    /// @param platformWallet Wallet address that will receive the platfrom fee.
+    /// @param platformFee Fee amount in wei the platfrom wallet is entitled to.
+    /// @param networkFee Fee amount (in wei) to be allocated for BRR
+    function handleFees(
+        IERC20 token,
+        address[] calldata rebateWallets,
+        uint256[] calldata rebateBpsPerWallet,
+        address platformWallet,
+        uint256 platformFee,
+        uint256 networkFee
+    ) external payable override onlyKyberNetwork {
+        require(token == ETH_TOKEN_ADDRESS, "token not eth");
+        require(msg.value == platformFee.add(networkFee), "msg.value not equal to total fees");
+
+        // handle platform fee
+        feePerPlatformWallet[platformWallet] = feePerPlatformWallet[platformWallet].add(
+            platformFee
+        );
+        totalPlatformFeeWei = totalPlatformFeeWei.add(platformFee);
+        emit HandleFee(ETH_TOKEN_ADDRESS, platformWallet, platformFee, rebateWallets, rebateBpsPerWallet, networkFee);
+
+        if (networkFee == 0) {
+            return;
+        }
+
+        (bool success, ) = address(this).call(
+            abi.encodeWithSignature(
+                "calculateAndRecordFeeData(address[],uint256[],uint256)",
+                rebateWallets,
+                rebateBpsPerWallet,
+                networkFee
+            )
+        );
+        if (!success) {
+            emit HandleFeeFailed(rebateWallets, rebateBpsPerWallet, networkFee);
+        }
+    }
+
+    function calculateAndRecordFeeData(
+        address[] calldata rebateWallets,
+        uint256[] calldata rebateBpsPerWallet,
+        uint256 feeBRRWei
+    ) external virtual {
+        require(msg.sender == address(this), "only Feehandler contract can call this function");
+        uint256 rebateWei = feeBRRWei.mul(rebateBps).div(BPS);
+        uint256 rewardWei = feeBRRWei.mul(rewardBps).div(BPS);
+
+        rebateWei = updateRebateValues(rebateWei, rebateWallets, rebateBpsPerWallet);
+
+        totalRewardWei = totalRewardWei.add(rewardWei);
+
+        uint burnAmountWei = feeBRRWei.sub(rewardWei).sub(rebateWei);
+
+        emit BRRFeeDistribution(
```

Comment:
Thinking again about this event.

it will be easier for us if here we have here an event that has all data.

so if this part doesn't fail (which it shouldn't) we can track all what we need by one event.

maybe existing feeDistributed from regular fee handler is suitable.

so if this parts works OK
we can use

Reply:
if the call for `calculateAndRecordFeeData` failed, then will we track the feeDistributed right? 
In other cases, we will handle as normally?